Fix documentation (md and inline) for C++ and Python spech samples

dorloff opened the pull request:
### Details:
 - Update and correct documentation for C++ and Python speech samples

### Tickets:
 - 94314

mdeisher reviewed (commented):
I have not found that the port number is required with the -i option.
For -o and -r, do we support npz too?
HPP, Line 17 should be "Path(s) to input file(s)."
Line 94:  do we really want to keep "dynamic" as an option since it is not implemented.
Line 97 should be "Number of weight bits" or "Weight resolution in bits"

dorloff commented:
> I have not found that the port number is required with the -i option.

Do you mean you think it's not required? I double checked, it is required.

> For -o and -r, do we support npz too?

Yes.

> HPP, Line 17 should be "Path(s) to input file(s)."

Fixed, thanks.

> Line 94: do we really want to keep "dynamic" as an option since it is not implemented.

Indeed, it makes no sense to keep it, removed.

> Line 97 should be "Number of weight bits" or "Weight resolution in bits"

FIxed, thanks.
